libs/perun/dialogs/src/lib/password-reset-dialog/password-reset-dialog.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  ApiRequestConfigurationService,
+  NotificatorService,
+  StoreService,
+} from '@perun-web-apps/perun/services';
+import { UsersManagerService } from '@perun-web-apps/perun/openapi';
+import { loginAsyncValidator } from '@perun-web-apps/perun/namespace-password-form';
+import { FormBuilder, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { CustomValidators } from '@perun-web-apps/perun/utils';
+import { PasswordLabels } from '@perun-web-apps/perun/models';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+export interface ResetPasswordDialogData {
+  namespace: string;
+  login: string;
+}
+
+@Component({
+  selector: 'perun-web-apps-password-reset-dialog',
+  templateUrl: './password-reset-dialog.component.html',
+  styleUrls: ['./password-reset-dialog.component.scss'],
+})
+export class PasswordResetDialogComponent implements OnInit {
+  loading = false;
+  success = false;
+  successMsg: string;
+  error = false;
+  errorMsg: string;
+  language = 'en';
+  newPasswdForm: FormGroup<{
+    passwordCtrl: FormControl<string>;
+    passwordAgainCtrl: FormControl<string>;
+  }>;
+  labels: PasswordLabels;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) private data: ResetPasswordDialogData,
+    private dialogRef: MatDialogRef<PasswordResetDialogComponent>,
+    private storeService: StoreService,
+    private translate: TranslateService,
+    private apiRequestConfiguration: ApiRequestConfigurationService,
+    private usersService: UsersManagerService,
+    private formBuilder: FormBuilder,
+    private notificator: NotificatorService
+  ) {
+    translate
+      .get('SHARED_LIB.PERUN.COMPONENTS.RESET_PASSWORD_DIALOG.SUCCESS')
+      .subscribe((m: string) => (this.successMsg = m));
```

Comment:
Please use `PerunTranslateService` instead. Also we usually don't save the translation since it's not reused. You can use the `instant()` method of the translator.